lib/src/model/over_the_board/over_the_board_clock.dart

```diff
@@ -93,6 +93,16 @@ class OverTheBoardClock extends _$OverTheBoardClock {
 
     state = state.copyWith(activeClock: newSideToMove);
   }
+
+  void giveTime(Duration amount) {
+    final newTime = state.timeLeft(state.activeClock!)! + amount;
```

Comment:
I assume the clock can be disabled in which case this method would crash. You should check for that first.